### PR TITLE
Fix tests and update nock

### PR DESCRIPTION
**Pull request by mehtaphysical**
Some tests where failing because of a bug in nock. Updating nock revealed a bunch of other failing tests.

### CODE REVIEW

File: test/commands/backups/capture.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const shouldCapture = function (cmdRun) {
   it('captures a db', () => {
     addon.app.name = 'myapp'
     api = nock('https://api.heroku.com')
-    api.post('/actions/addon-attachments/resolve', {app: 'myapp', addon_attachment: 'DATABASE_URL'}).reply(200, [{addon}])
+    api.post('/actions/addon-attachments/resolve', {app: 'myapp', addon_attachment: 'DATABASE_URL', addon_service: 'heroku-postgresql'}).reply(200, [{addon}])
```
Comment (keiko713) [inline test/commands/backups/capture.js:33]: Can you elaborate more on the bug of nock? Also, why we now need to use this `addon_service`?

Comment (mehtaphysical) [reply test/commands/backups/capture.js:33]: 1) The bug in nock wasn't properly overriding the http request method. So, for example, `test/commands/backups/download.js` was failing:

```
  1) pg:backups:download with no id downloads to latest.dump:
     UnexpectedError:
expected '{"id":"not_found","message":"Not found."}' to equal '{}'

{"id":"not_found","message":"Not found."}
{}

      at cmd.run.then (test/commands/backups/download.js:35:19)
      at <anonymous>
      set UNEXPECTED_FULL_TRACE=true to see the full stack trace

  2) pg:backups:download with id downloads to latest.dump:
     UnexpectedError:
expected '{"id":"not_found","message":"Not found."}' to equal '{}'

{"id":"not_found","message":"Not found."}
{}

      at cmd.run.then (test/commands/backups/download.js:42:19)
      at <anonymous>
      set UNEXPECTED_FULL_TRACE=true to see the full stack trace
```

I think it's this PR in particular that fixes this: https://github.com/node-nock/nock/pull/929.

2) Nock checks to make sure the post body matches the expected body. In the version we are currently using this check isn't enforced (see this issue: https://github.com/node-nock/nock/issues/920). So the `addon_service` is getting posted and we need to add it to our expected post body.

**Comment by mehtaphysical** [comment]
This will be fixed by #173 . I'm going to close this

**Comment by keiko713** [comment]
Oh sad, thanks for the explanation though! Also maybe you wanna delete a branch ;)

**Pull request closed without merging**